src/components/Pagination/Pagination.js

```diff
@@ -0,0 +1,164 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import bem from 'bem';
+import styles from './Pagination.scss';
+import Button from '../Button';
+import PaginationButton from './PaginationButton';
+
+const { block, elem } = bem({
+    name: 'Pagination',
+    classnames: styles,
+    propsToMods: ['align']
+});
+
+const Pagination = props => {
+    const {
+        align,
+        currentPage,
+        maxPageButtons,
+        totalPages,
+        prevLabel,
+        nextLabel,
+        lastLabel,
+        firstLabel,
+        onClick,
+        ...rest
+    } = props;
+
+    const defineRange = () => {
+        const showBefore = Math.floor((maxPageButtons - 1) / 2);
+        const start = Math.max(
+            2,
+            Math.min(currentPage - showBefore, totalPages - maxPageButtons + 1)
+        );
+        const end = Math.min(totalPages, start + maxPageButtons - 2);
```

Comment:
` - 2` means the last button is always missing.

`currentPage={8} totalPages={8} maxPageButtons={7}`

Reply:
works ok now